src/include/pmix_globals.h

```diff
@@ -351,6 +352,7 @@ PMIX_CLASS_DECLARATION(pmix_info_caddy_t);
     do {                                        \
         while ((a)) {                           \
             usleep(10);                         \
+            PMIX_ACQUIRE_OBJECT((a));           \
```

Comment:
The `PMIX_ACQUIRE_OBJECT` should be just outside of the loop. We don't need a memory barrier on `(a)` since it's volatile (and if it takes a while for the cache to flush it's not a big slowdown). Before we leave the macro though we need to do the `PMIX_ACQUIRE_OBJECT` once to force the barrier.